### PR TITLE
fix:issue#1212 candidate with empty string

### PR DESCRIPTION
**Pull request by scorpionknifes**
#### Description

based on https://github.com/pion/webrtc/pull/1428#issuecomment-695911422

merge after pion/ice accept nil candidate https://github.com/pion/ice/pull/315#issuecomment-739858837

thx @DevRockstarZ for PR https://github.com/pion/webrtc/pull/1428

#### Reference issue
Fixes https://github.com/pion/webrtc/issues/1212

### CODE REVIEW

**Comment by Sean-Der** [comment]
@scorpionknifes this looks great to me! I would be happy to take this today :) 

* Can you squash this down into one commit
* I would make you the author, and then use `Co-authored-by:`
* You have a few linter errors

**Comment by Sean-Der** [comment]
@scorpionknifes `pion/ice@v2.0.14` is tagged!

**Comment by scorpionknifes** [comment]
I'm gonna need to take a deeper look

**Comment by Sean-Der** [comment]
If think if you move your test into `peerconnection_go_test.go` it will work! It is just being run for WASM also currently.

File: peerconnection.go [hunk 1]
```diff
@@ -1454,13 +1454,28 @@ func (pc *PeerConnection) RemoteDescription() *SessionDescription {
...
 	candidateValue := strings.TrimPrefix(candidate.Candidate, "candidate:")
+
+	if candidateValue == "" {
+		pc.iceTransport.lock.RLock()
+		defer pc.iceTransport.lock.RUnlock()
+
+		agent := pc.iceTransport.gatherer.getAgent()
+		if agent == nil {
+			return fmt.Errorf("%w: unable to add remote candidates", errICEAgentNotExist)
+		}
```
Comment (scorpionknifes) [inline peerconnection.go:1474]: Anyway, I can cover this by setting agent=nil in test?

**Comment by Sean-Der** [comment]
Merged with 3eb51d8a091a98c948654876bc665afe2b0368b1

Fantastic work @scorpionknifes really appreciate all the work you have been doing lately? Do you have a Twitter? I would love to give you a shout out for all the great work you are doing (if not will just do your GitHub)

**Pull request closed without merging**